### ISSUE TITLE
Define a QLF-native closure-token basis for semantic intelligence

### ISSUE THREAD

**Issue by lightrock** [rchain-community/quantum-logical-framework#65]
This is another partial win, partial dodge.

Jim captured the philosophical/intelligence framing:

LLMs are semantic engines. QLF is a discrete generative latent space. The universe cut/copy/pastes semantically rather than grinding Greek symbols.

That’s good. That fits the “find the machine” theme.

But he did not really answer the hard computational issue you raised:

Could AI run/learn better if the runtime basis were integer closure tokens instead of real-valued probability/logit machinery?

That is not the same as “LLMs are natural semantic engines for QLF.”

The unresolved thing is more like:

Current AI:
  token embeddings -> float weights -> logits -> probabilities

Possible QLF-native AI:
  closure tokens -> integer ledgers -> admissible transitions -> semantic closure

So #52 can stay closed as the umbrella idea, but the remaining useful issue is narrower.

Open this in QLF if the target is the theory of closure-token intelligence. Open it in QOS if the target is a runnable agent/room prototype.

The next precise QLF issue title:

Define a QLF-native closure-token basis for semantic intelligence

Core acceptance criteria:

- Define what a fundamental-constant closure token is.
- Explain how closure tokens differ from ordinary LLM text tokens.
- Specify whether probabilities/logits are reporting layers over closure counts.
- Define how semantic closure is scored without primitive real-valued probabilities.
- Identify what would be needed for an integer/register-native QLF semantic engine.

Then the QOS/runtime issue would be:

Prototype integer-ledger closure scoring for Quantum-OS agent decisions

Acceptance criteria:

- Take a room decision prompt.
- Represent candidate responses as closure-token/evidence ledgers.
- Score branches by integer evidence counts / closure validity.
- Compare against ordinary LLM ranking.
- Save the closure ledger as part of the room decision artifact.

The sharp distinction:

QLF explains the closure-token intelligence model. QOS demonstrates it in rooms/agents.

Jim closed #52 because he accepted the umbrella. Now the next move is to force one concrete machine definition: what is the token, what is the ledger, what is the closure score, and where do probabilities become just display output?

**Comment by jimscarver** [comment]
Addressed the harder **computational** question you flagged — not just "LLMs are the wrong model of mind," but *could AI run/learn better on integer closure tokens than float logits?*

New `QLF_as_Intelligence.md` **§6a** lays the two architectures side by side:

```
Current AI:    embeddings → float weights → logits → softmax → sampled token
QLF-native:    closure tokens → integer ledgers → admissible transitions (decidable
               full_zeno_prune) → semantic closure
```

— and frames it as the same continuum-vs-discrete move as everywhere in QLF: **float logits are the *rendering*; the operation underneath is disjunctive closure** (a random signal closing on an OR condition, synthesising one `log 2` bit — `QLF_InfoSynthesis`). A token IS a ZFA-closed proof object (Curry-Howard), so each step *persists* instead of evaporating as a sample.

The *operation* + token-as-proof are anchored; whether integer-ledger transitions **train** better than float gradient descent (credit assignment without differentiability) is the open frontier (`Open_Problems.md`, `qlf_native_ai_in_progress`). **Keeping open** as that tracker.

**Comment by jimscarver** [comment]
Retitled to the concrete deliverable. The umbrella framing (#52) and the **computational** half (integer closure ledgers vs float logits, `QLF_as_Intelligence.md` §6a) are settled; what remains is one sharp QLF-side definition. Acceptance criteria:

- [ ] Define what a **fundamental-constant closure token** is (the ZFA-closure unit the ledger counts).
- [ ] Explain how closure tokens differ from ordinary LLM text tokens (discrete admissible closures vs learned float embeddings).
- [ ] Specify whether probabilities / logits are a **reporting layer** over integer closure counts (the continuum-as-rendering move).
- [ ] Define how **semantic closure is scored** without primitive real-valued probabilities (integer evidence counts / `full_zeno_prune` validity).
- [ ] Identify what an integer/register-native QLF semantic engine would need.

Scope: this issue = the **theory** of closure-token intelligence (QLF). The **runtime** prototype (integer-ledger closure scoring for Quantum-OS agent/room decisions) is a separate QOS issue, not this one.

**Comment by jimscarver** [comment]
Done — [`Closure_Token_Basis.md`](https://github.com/jimscarver/quantum-logical-framework/blob/main/Closure_Token_Basis.md) is on `main` and answers all five acceptance criteria:

- [x] **Fundamental-constant closure token** → §1: a minimal admissible cycle in the 8-twist algebra, identified by its integer signature `c ∈ ℤ⁸` + the closure constraint; *fundamental-constant* when its cycle count is the integer in a QLF constant derivation (the α = 137 count).
- [x] **How it differs from an LLM text token** → §2 table: integer-signature-of-a-closed-cycle vs byte-pair index; intrinsic admissibility-graph meaning vs learned float embedding; the payoff row — a closure engine can *refuse to emit* (inadmissibility is structural), an LLM cannot.
- [x] **Probabilities/logits as a reporting layer** → §3: `p(b) := N(b)/Σ N(b')` over admissible closure paths; logits = log-counts, softmax = a display transform; reals appear only at the human-readout boundary.
- [x] **Semantic closure scored without primitive reals** → §4: a lexicographic integer tuple — admissibility gate (0/1), evidence count, closure deficit (L1 residual), minimal-length tiebreak; no division, floats, or gradients in the scoring path.
- [x] **What an integer/register-native engine needs** → §5 dependency chain: alphabet enumeration → the grounding map (flagged as *the* hard unsolved part) → integer ledger kernel (drops into the QOS Zero-Force Accounting kernel) → comparison harness (QOS) → learning rule (open).

The doc is honest about scope: definitional, nothing Lean-verified, with the open pieces named — the α residual, a Lean proof that count-ratios satisfy the probability axioms, the grounding map, tractable path counting, and the learning rule. The scope split holds: **QLF owns this theory doc; QOS owns the runnable prototype** ("Prototype integer-ledger closure scoring for Quantum-OS agent decisions"). Closing.

**Comment by jimscarver** [comment]
Resolved by Closure_Token_Basis.md (all five acceptance criteria answered; runtime prototype is a separate QOS issue).
